app/components/provider_interface/user_org_permission_breakdown_component.rb

```diff
@@ -0,0 +1,46 @@
+module ProviderInterface
+  class UserOrgPermissionBreakdownComponent < ViewComponent::Base
+    attr_reader :provider, :permission
+
+    def initialize(provider:, permission:)
+      @provider = provider
+      @permission = permission
+    end
+
+    def permitted_by_providers
+      @permitted_by_providers ||= Provider.where(id: training_providers_with_permissions.pluck(:id))
+                                          .or(Provider.where(id: ratifying_providers_with_permissions.pluck(:id)))
+    end
+
+    def prohibited_by_providers
+      @prohibited_by_providers ||= Provider.where(id: training_providers_without_permissions.pluck(:id))
+                                           .or(Provider.where(id: ratifying_providers_without_permissions.pluck(:id)))
+    end
+
+  private
+
+    def training_providers_with_permissions
+      Provider.joins(:training_provider_permissions)
+              .where(provider_relationship_permissions: { ratifying_provider: provider,
+                                                          "ratifying_provider_can_#{permission}" => true })
```

Comment:
Since we're only using a list of provider ids that come from this query, I think we could make this type of method more readable:
```
def ratifying_partner_ids_for_which_permission_applies
  provider.training_provider_permissions.where("training_provider_can_#{permission}": true).pluck(:ratifying_provider_id)
end
```

Reply:
What do you think about the above logic @despo ? It's a bit clearer that we are returning e.g. a ratifying provider id